src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -2039,6 +2039,10 @@ private static void UpdateFormBindingArgumentExpressions(RequestDelegateFactoryC
                 formDict,
                 Expression.Constant(CultureInfo.InvariantCulture),
                 Expression.Call(AsMemoryMethod, formBuffer, Expression.Constant(0), Expression.Constant(formDataMapperOptions.MaxKeyBufferSize))));
+        // name_reader.MaxRecursionDepth = formDataMapperOptions.MaxRecursionDepth;
```

Comment:
I didn't see an issue or code for this in RDG

Reply:
RDG doesn't support the complex form binding feature because the core logic relies on reflection and dynamic code gen (see [this for example](https://github.com/dotnet/aspnetcore/blob/83826127edc34806995b85b214d1d4591eb382b0/src/Components/Endpoints/src/FormMapping/Factories/ComplexType/ComplexTypeExpressionConverterFactoryOfT.cs)). There's a goal to move this to use source generators for .NET 9 (or later) but for now I feel comfortable with bringing this goodness to users not leveraging publish AoT.